pip/pep425tags.py

```diff
@@ -198,7 +247,8 @@ def get_supported(versions=None, noarch=False):
                 supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
         # Has binaries, does not use the Python API:
-        supported.append(('py%s' % (versions[0][0]), 'none', arch))
+        for arch in arches:
+            supported.append(('py%s' % (versions[0][0]), 'none', arch))
```

Comment:
I also think it is a bug fix, and for a bug that was never noticed before because it only affects wheels that combine a `none` ABI tag + a non-trivial platform tag, which is a rare (but totally valid) configuration.